server/jetstream_benchmark_test.go

```diff
@@ -1174,3 +1174,161 @@ func BenchmarkJetStreamKV(b *testing.B) {
+		replicaAndClusterSize []int
+		numKeys               []int
+		objectSizeRange       [][]int
+		rwRatio               []float64
+		storageType           []nats.StorageType
+	}{
+		[]int{1, 3},             // 1 replica, 3 replicas
+		[]int{10, 1000, 100000}, // 10 keys, 1K keys, 100K keys
+		[][]int{
+			{1024, 1024}, // 1KB byte objects
+			//{102400, 102400}, // 1MB byte objects
+		},
+		[]float64{1.0, 0.0, 0.8},                                 // readOnly, writeOnly, 80% read
+		[]nats.StorageType{nats.MemoryStorage, nats.FileStorage}, // memory storage, file storage
+	}
+
+	// run benchmarks using options matrix
+	for _, clusterSize := range benchmarkOptions.replicaAndClusterSize {
+		replicas := clusterSize
+		for _, numKeys := range benchmarkOptions.numKeys {
+			for _, objectSizeRange := range benchmarkOptions.objectSizeRange {
+				minBytes, maxBytes := objectSizeRange[0], objectSizeRange[1]
+				for _, rwRatio := range benchmarkOptions.rwRatio {
+					var rwRatioType string
+					switch rwRatio {
+					case 1.0:
+						rwRatioType = "readOnly"
+					case 0.0:
+						rwRatioType = "writeOnly"
+					default:
+						rwRatioType = fmt.Sprintf("rwRatio=%0.2f%%", rwRatio)
+					}
+					for _, storageType := range benchmarkOptions.storageType {
+						bName := fmt.Sprintf(
+							"N=%d,R=%d,K=%d/storageType=%s/minBytes=%db,maxBytes=%db/%s",
+							clusterSize,
+							replicas,
+							numKeys,
+							storageType.String(),
+							minBytes,
+							maxBytes,
+							rwRatioType,
+						)
+						b.Run(
+							bName,
+							func(b *testing.B) {
+								if verbose {
+									b.Logf("Running %s workload %s with %d messages", bName, bName, b.N)
+								}
+
+								if verbose {
+									b.Logf("Setting up %d nodes", replicas)
+								}
+								var connectURL string
+								if clusterSize == 1 {
+									s := RunBasicJetStreamServer(b)
+									defer s.Shutdown()
+									connectURL = s.ClientURL()
+								} else {
+									cl := createJetStreamClusterExplicit(b, "BENCH_OBJ_STORE", clusterSize)
+									defer cl.shutdown()
+									cl.waitOnClusterReadyWithNumPeers(replicas)
+									cl.waitOnLeader()
+									// connect to leader and not replicas
+									connectURL = cl.leader().ClientURL()
+								}
+								nc, js := jsClientConnectURL(b, connectURL)
+								defer nc.Close()
+
+								// Initialize object store
+								objStoreName := fmt.Sprintf("%sTEST", objStoreNamePrefix)
+								if verbose {
+									b.Logf("Creating ObjectStore %s with R=%d", objStoreName, replicas)
+								}
+								objStoreConfig := &nats.ObjectStoreConfig{
+									Bucket:   objStoreName,
+									Replicas: replicas,
+									Storage:  storageType,
+								}
+								objStore, err := js.CreateObjectStore(objStoreConfig)
+								if err != nil {
+									b.Fatalf("Error creating ObjectStore: %v", err)
+								}
+
+								// Initialize keys
+								if initKeys {
+									rng := rand.New(rand.NewSource(int64(seed)))
```

Comment:
You are initializing multiple `rng`s with the same seed (here and below).
Just initialize one per experiment at the top of the run method